eslint.config.js

```diff
@@ -58,7 +58,7 @@ export default [
 	// TypeScript
 	...tseslint.config({
 		files: ["**/*.ts"],
-		extends: [...tseslint.configs.strict, ...tseslint.configs.stylistic],
+		extends: [...tseslint.configs.strict],
```

Comment:
😄 not a fan of the stylistic config? Any reason in particular to remove this?

From running locally, I see two complaints:

```plaintext
/Users/josh/repos/rewrite/packages/core/src/types.ts
  108:8   error  A record is preferred over an index signature                                        @typescript-eslint/consistent-indexed-object-style
  389:12  error  Array type using 'Array<SuggestedEdit>' is forbidden. Use 'SuggestedEdit[]' instead  @typescript-eslint/array-type
```

* [`@typescript-eslint/array-type`](https://typescript-eslint.io/rules/array-type): purely stylistic for consistency. If you prefer `Array<...>` that's available as an option.
* [`@typescript-eslint/consistent-indexed-object-style`](https://typescript-eslint.io/rules/consistent-indexed-object-style): it's meant to be purely stylistic, but there are some changes between interfaces and records. If you want to use catch-all string signatures the way `RuleVisitor` is set up now, I'd just disable the rule with a comment. Either inline or in the config altogether.

Reply:
Because I didn't think either of these warnings were helpful, and I didn't feel like fighting other rules that might come up (generally not a fan of applying rules blindly, especially ones that just seem to be enforcing stylistic preferences that I don't have).

Reply:
Understood. My (very obviously biased as typescript-eslint maintainer) recommendation would be to keep them enabled, but I understand why you'd want to avoid the stylistic preferences.

The terminology here is unfortunate: "stylistic" as used by that shareable config encompasses more than what "stylistic" means in ESLint core. It contains more than just consistency rules; it also has some real best practice violations and checks for likely bug causing syntax.

I'd recommend at least considering leaving the following rules (here or as a followup issue) that most folks see as definitive positives beyond just stylistic consistency:

* [`@typescript-eslint/adjacent-overload-signatures`](https://typescript-eslint.io/rules/adjacent-overload-signatures)
* [@typescript-eslint/consistent-type-definitions](https://typescript-eslint.io/rules/consistent-type-definitions)
* [@typescript-eslint/no-confusing-non-null-assertion](https://typescript-eslint.io/rules/no-confusing-non-null-assertion)
* [@typescript-eslint/no-inferrable-types](https://typescript-eslint.io/rules/no-inferrable-types)

https://typescript-eslint.io/rules/?=stylistic-xtypeInformation has the full listing of rules in `tseslint.configs.stylistic`.

Reply:
Understood. My (very obviously biased as typescript-eslint maintainer) recommendation would be to keep them enabled, but I understand why you'd want to avoid the stylistic preferences.

The terminology here is unfortunate: "stylistic" as used by that shareable config encompasses more than what "stylistic" means in ESLint core. It contains more than just consistency rules; it also has some real best practice violations and checks for likely bug causing syntax.

I'd recommend at least considering leaving the following rules (here or as a followup issue) that most folks see as definitive positives beyond just stylistic consistency:

* [`@typescript-eslint/adjacent-overload-signatures`](https://typescript-eslint.io/rules/adjacent-overload-signatures)
* [`@typescript-eslint/consistent-type-definitions`](https://typescript-eslint.io/rules/consistent-type-definitions)
* [`@typescript-eslint/no-confusing-non-null-assertion`](https://typescript-eslint.io/rules/no-confusing-non-null-assertion)
* [`@typescript-eslint/no-inferrable-types`](https://typescript-eslint.io/rules/no-inferrable-types)

https://typescript-eslint.io/rules/?=stylistic-xtypeInformation has the full listing of rules in `tseslint.configs.stylistic`.

Reply:
Understood. My (very obviously biased as typescript-eslint maintainer) recommendation would be to keep them enabled, but I understand why you'd want to avoid the stylistic preferences.

The terminology here is unfortunate: "stylistic" as used by that shareable config encompasses more than what "stylistic" means in ESLint core. It contains more than just consistency rules; it also has some real best practice violations and checks for likely bug causing syntax.

I'd recommend at least considering leaving the following rules (here or as a followup issue) that most folks see as definitive positives beyond just stylistic consistency:

* [`@typescript-eslint/adjacent-overload-signatures`](https://typescript-eslint.io/rules/adjacent-overload-signatures)
* [`@typescript-eslint/no-confusing-non-null-assertion`](https://typescript-eslint.io/rules/no-confusing-non-null-assertion)
* [`@typescript-eslint/no-inferrable-types`](https://typescript-eslint.io/rules/no-inferrable-types)

...and similarly at least considering these rules on _"just pick one and stick with it"_ consistency points that, in my experience, tend to get annoyingly split in a repo over time:

* [`@typescript-eslint/array-type`](https://typescript-eslint.io/rules/array-type)
* [`@typescript-eslint/consistent-type-definitions`](https://typescript-eslint.io/rules/consistent-type-definitions)

https://typescript-eslint.io/rules/?=stylistic-xtypeInformation has the full listing of rules in `tseslint.configs.stylistic`.

Reply:
Okay, I added it back.